Fix some grammar/typos (#17)
HKalbasi/zngur: book/src/how_it_works.md

## How a Rust type is represented in C++

A normal rust type ABI is undefined, so passing it directly in a cross-language function call is undefined behavior. The
thing that is guaranteed is that the size and alignment of a type won't change during a compile session. By adding static assertions
against the user provided size and alignment in the `main.zng` file, Zngur ensures that it knows the correct size and alignment of the
type for this compile session. Knowing the size and alignment of a type enables `std::ptr::read` and `std::ptr::write`. These functions
only need the pointer to be valid (which basically means `ptr..ptr+size` should belong to a single live chunk of memory,
[read more](https://doc.rust-lang.org/std/ptr/index.html#safety)) and

> aligned. So Zngur can use the pointer to the underlying `data` in those functions:

Reviewer: here I meant the `data` mentioned in line 17 (below). But if you think it is now clearer, I'm fine with the change.

> aligned. So Zngur can use the pointer to the underlying `data` in those functions:

Author: Ah, I misunderstood. How about this?

```suggestion
aligned. So Zngur can use a pointer to `data`, as defined below, in those functions:
```
